core/src/main/java/com/graphhopper/routing/template/ViaRoutingTemplate.java

```diff
@@ -119,8 +120,14 @@ public ViaRoutingTemplate(GHRequest ghRequest, GHResponse ghRsp, LocationIndex l
 
             int idx = 0;
             for (Path path : tmpPathList) {
-                if (path.getTime() < 0)
-                    throw new RuntimeException("Time was negative " + path.getTime() + " for index " + idx + ". Please report as bug and include:" + ghRequest);
+                if (path.getTime() < 0) {
+                    Map<String, Object> details = new HashMap<>();
+                    details.put("time", path.getTime());
+                    details.put("from", queryResults.get(idx).getSnappedPoint());
+                    details.put("to", queryResults.get(idx + 1).getSnappedPoint());
```

Comment:
All these details could be included in the message, there is no need for an automated way to read this as this is a bug and should not occur. Also will it still result in a 500 http code? This is very important.